Sources/SnapshotTesting/Common/View.swift

```diff
@@ -736,34 +736,35 @@ func renderer(bounds: CGRect, for traits: UITraitCollection) -> UIGraphicsImageR
 }
 
 private func add(traits: UITraitCollection, viewController: UIViewController, to window: UIWindow) {
-  let rootViewController = UIViewController()
-  rootViewController.view.backgroundColor = .clear
-  rootViewController.view.frame = window.frame
-  rootViewController.view.translatesAutoresizingMaskIntoConstraints =
-    viewController.view.translatesAutoresizingMaskIntoConstraints
-  rootViewController.preferredContentSize = rootViewController.view.frame.size
-
-  rootViewController.addChild(viewController)
-  viewController.view.frame = rootViewController.view.frame
-  rootViewController.view.addSubview(viewController.view)
-
-  if viewController.view.translatesAutoresizingMaskIntoConstraints {
-    viewController.view.autoresizingMask = [.flexibleWidth, .flexibleHeight]
-  } else {
-    NSLayoutConstraint.activate([
-      viewController.view.topAnchor.constraint(equalTo: rootViewController.view.topAnchor),
-      viewController.view.bottomAnchor.constraint(equalTo: rootViewController.view.bottomAnchor),
-      viewController.view.leadingAnchor.constraint(equalTo: rootViewController.view.leadingAnchor),
-      viewController.view.trailingAnchor.constraint(equalTo: rootViewController.view.trailingAnchor),
+  let rootViewController: UIViewController
+  if viewController != window.rootViewController {
+    rootViewController = UIViewController()
+    rootViewController.view.backgroundColor = .clear
+    rootViewController.view.frame = window.frame
+    rootViewController.view.translatesAutoresizingMaskIntoConstraints =
+      viewController.view.translatesAutoresizingMaskIntoConstraints
+    rootViewController.preferredContentSize = rootViewController.view.frame.size
+    viewController.view.frame = rootViewController.view.frame
+    rootViewController.view.addSubview(viewController.view)
+    if viewController.view.translatesAutoresizingMaskIntoConstraints {
+      viewController.view.autoresizingMask = [.flexibleWidth, .flexibleHeight]
+    } else {
+      NSLayoutConstraint.activate([
+        viewController.view.topAnchor.constraint(equalTo: rootViewController.view.topAnchor),
+        viewController.view.bottomAnchor.constraint(equalTo: rootViewController.view.bottomAnchor),
+        viewController.view.leadingAnchor.constraint(equalTo: rootViewController.view.leadingAnchor),
+        viewController.view.trailingAnchor.constraint(equalTo: rootViewController.view.trailingAnchor),
       ])
+    }
+    rootViewController.addChild(viewController)
+    rootViewController.setOverrideTraitCollection(traits, forChild: viewController)
```

Comment:
Should traits still be added to either VC? It might be surprising to pass traits to the strategy and not have them reflected.